accounts/views.py

```diff
@@ -1,3 +1,38 @@
-from django.shortcuts import render
+from django.db.models import Count
+from rest_framework.viewsets import ModelViewSet, GenericViewSet
+from rest_framework.mixins import CreateModelMixin, ListModelMixin, RetrieveModelMixin
+from rest_framework.permissions import IsAuthenticated, IsAdminUser
 
-# Create your views here.
+from accounts.models import *
+from .serializers import *
+from .models import *
+
+
+"""Api View for listing all users and retrieving specific users using their id's"""
+class UserViewSet(ListModelMixin, RetrieveModelMixin, GenericViewSet):
+    serializer_class = UserSerializer
+    permission_classes = [IsAdminUser]
+
+    def get_queryset(self):
+        if self.request.method == 'GET':
+            return User.objects.annotate(number_of_accounts=Count('account'))#Adds a new column that counts the number of accounts.
+        return User.objects.all()
+
+
+"""Api view to be used when a user first registers to the system"""
+class RegisterAccountViewSet(CreateModelMixin, RetrieveModelMixin, ListModelMixin, GenericViewSet):
```

Comment:
Preferably, create  Retrieve view set separately from the Register Account View set. A retrieve or list view isn't required at a Register account viewset